### PR TITLE
Move some internal scheduler enums to libcrmcommon

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@wenningerk , can you review this one when you get a chance? Low priority, thanks

File: include/crm/pengine/internal.h [hunk 1]
```diff
@@ -138,32 +138,11 @@ pe_resource_t *pe__last_group_member(const pe_resource_t *group);
...
-    pe_wo_upstart       = (1 << 13),
-    pe_wo_nagios        = (1 << 14),
-    pe_wo_set_ordering  = (1 << 15),
-};
-
 extern uint32_t pe_wo;
 
 #define pe_warn_once(pe_wo_bit, fmt...) do {    \
         if (!pcmk_is_set(pe_wo, pe_wo_bit)) {  \
-            if (pe_wo_bit == pe_wo_blind) {     \
```
Comment (wenningerk) [inline include/crm/pengine/internal.h:183]: This set of patches anyway creates a bit of an inconsistent result (should we merge it like that?) with the enums being the new names and everything to work with them still on the old pattern but pe_wo_bit could already go to pcmk__wo_bit already just  for better readability.

**Comment by wenningerk** [comment]
Doesn't make it worse than before but having enums/flags and strings and whatnot with the same naming scheme is a bit confusing. Would it be e.g. useful to add flg to the enum naming scheme as to be able to distinguish or is the more compact code that reads more like everyday-language the stronger argument?

File: lib/pengine/failcounts.c [hunk 2]
```diff
@@ -388,7 +388,7 @@ pe_get_failcount(const pe_node_t *node, pe_resource_t *rsc,
     }
```
Comment (wenningerk) [inline lib/pengine/failcounts.c:389]: There is still a  pe_fc_flags_e leftover in the comments above.

File: include/crm/common/digests_internal.h [hunk 3]
```diff
@@ -16,6 +16,14 @@
 extern "C" {
 #endif
 
+// Digest comparison results
+enum pcmk__digest_result {
+    pcmk__digest_unknown,   // No digest available for comparison
```
Comment (wenningerk) [inline include/crm/common/digests_internal.h:21]: ok ... that all stays strictly locally on one node meaning the actual values shouldn't matter.
having a zeroed struct default to unknown is definitely the more robust approach
but telling if the previous default to match was used or happened by error (in any way
changing behavior) isn't that trivial ... try to do my best ...

**Comment by wenningerk** [comment]
Looking good!
The actual introduction and use of the enums should be trivial and well checked by the compiler -
unless the subtle details are below my understanding level ;-)
So my comments are of a more general nature. Nothing at least should change behavior as far
as I can see.

File: include/crm/pengine/internal.h (hunk 1 above)
Comment (kgaillot) [reply include/crm/pengine/internal.h:183]: Good point, this macro is ported and updated in a later commit. I was able to move up one related commit pretty easily, so I'll push that one up, but the one for this depends on a bunch of others, and I'd rather not try to split it now. Hopefully it won't be too long ...

**Comment by kgaillot** [comment]
> Doesn't make it worse than before but having enums/flags and strings and whatnot with the same naming scheme is a bit confusing. Would it be e.g. useful to add flg to the enum naming scheme as to be able to distinguish or is the more compact code that reads more like everyday-language the stronger argument?

From the caller's perspective, these are exclusive values rather than flags, but I can do `enum pcmk__sched_warnings`

**Comment by kgaillot** [comment]
> Loosing the pe-context by using the more general pcmk-prefix is probably not enough of an issue as to go more bulky with some additional notation.

Yep, I'm gradually moving toward `pcmk` for everything (like `g` for glib or `xml` for libxml2). Currently we're using `pcmk` for libcrmcluster, libcrmcommon, and libpacemaker. libpe_rules, libpe_status, and libcib will eventually be merged into libcrmcommon. That leaves libcrmservice, liblrmd, and libstonithd, and I'm not sure what will happen to them yet.

I've been using `pcmk_sched_` for scheduler-related stuff when it needs distinguishing. It's a good question whether `pe_warn_once()` should get it (in the later commit). `pcmk__sched_warn_once()` would be kind of long (`pe_proc_warn()` will become `pcmk__sched_warn()`), but I could go with that if you think it's worth it.

File: include/crm/common/digests_internal.h (hunk 3 above)
Comment (kgaillot) [reply include/crm/common/digests_internal.h:21]: Yep that was the only place I saw where it was allocated

**Comment by kgaillot** [comment]
Updated per review (including one new commit)

**Comment by wenningerk** [comment]
changes are looking good

**Pull request merged**